nio_cli/commands/base.py

```diff
@@ -1,13 +1,68 @@
+import requests
+
+
 class Base(object):
 
     def __init__(self, options, *args, **kwargs):
         self.options = options
         self.args = args
         self.kwargs = kwargs
         self._base_url = "http://{}:{}/{{}}".format(self.options['--ip'],
-                                                  self.options['--port'])
-        # TODO: don't always use Admin/Admin
+                                                    self.options['--port'])
+        # default to Admin/Admin
         self._auth = ('Admin', 'Admin')
 
     def run(self):
         raise NotImplementedError('Implement run() in the command.')
+
+    def get(self, *args, **kwargs):
+        return self._execute_request(
+            requests.get, *args, auth=self._get_credentials(), **kwargs)
+
+    def post(self, *args, **kwargs):
+        return self._execute_request(
+            requests.post, *args, auth=self._get_credentials(), **kwargs)
+
+    @staticmethod
+    def _execute_request(fn, *args, **kwargs):
+        response = fn(*args, **kwargs)
+        if response.status_code >= 400:
+            msg = "Client error, status: {}, message: {}".format(
+                response.status_code, response.reason)
+            print(msg)
+            raise RuntimeError(msg)
+
+        return response
+
+    def _get_credentials(self):
+        # allow to override last credentials set
+        (username, password) = self._auth
+
+        if "--username" in self.options and self.options["--username"]:
+            username = self.options.get("--username")
+        else:
```

Comment:
Does this `else` ever get hit? Based on other code it seems like `self.options.get("--username")` will return the default of `Admin` every time, regardless of whether `--username` was included or not.

Reply:
You're right. I'll go back and try to rework this logic. Do you think it should ever prompt for a username/password? I feel that if any command ever recieved a `401` from nio it could just display an auth error and reccomend running the command with set `--username` and `--password` options

Reply:
Yeah I think that makes way more sense. In general I like to avoid prompts in CLIs because doing them "properly" means adding flags to disable them, checking if there's a TTY, and all kinds of other baggage. The only time I like them, ironically, is for password inputs so you don't have to have your password in your shell history.